provider/github/internal/builtin/request.go

```diff
@@ -96,24 +106,36 @@ func (r Request) argsToRequest(terms []*ast.Term) (*retryablehttp.Request, error
 		return nil, fmt.Errorf("only GET requests are supported, got '%s'", method)
 	}
 
-	pathParams := r.parsePathParams(path)
+	u, err := buildRequestUrl(path, data)
+	if err != nil {
+		return nil, err
+	}
 
-	for _, p := range pathParams {
-		v, err := r.valueToString(data[p])
-		if err != nil {
-			return nil, err
-		}
+	return r.Client.NewRequest(method, u, nil)
+}
+
+func buildRequestUrl(path string, data map[string]any) (string, error) {
+	pathParams := parsePathParams(path)
+
+	if len(pathParams) != 0 {
```

Comment:
I believe this condition is not necessary. If `pathParams` is empty the `for` isn't executed anyway.